flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -37,6 +37,24 @@
 
 	private TaskExecutorResourceUtils() {}
 
+	// ------------------------------------------------------------------------
+	//  Generating JVM Parameters
+	// ------------------------------------------------------------------------
+
+	public static String generateJvmParametersStr(final TaskExecutorResourceSpec taskExecutorResourceSpec) {
+		final MemorySize jvmHeapSize = taskExecutorResourceSpec.getFrameworkHeapSize()
+			.add(taskExecutorResourceSpec.getTaskHeapSize())
+			.add(taskExecutorResourceSpec.getOnHeapManagedMemorySize());
+		final MemorySize jvmDirectSize = taskExecutorResourceSpec.getTaskOffHeapSize()
```

Comment:
What about `getOffHeapManagedMemorySize`?

Reply:
That uses unsafe memory, which should not be accounted for either `-Xmx` nor` -XX:MaxDirectMemorySize`.

Reply:
Good point. Thanks for the clarification.